vehicle/mkv/software/throttle/throttle.c

```diff
@@ -1,738 +1,189 @@
+    }
+    
+    if (deviation_coutner > 100) {
+        gFlag |= _BV(FLAG_IMPLAUSIBILITY); 
+    } else {
+        gFlag &= ~_BV(FLAG_IMPLAUSIBILITY);
+    }
 
-	// sprintf(uart_buf, "NEW: %u", new);
-	// LOG_println(uart_buf, strlen(uart_buf));
-	// sprintf(uart_buf, "OLD: %u", old);
-	// LOG_println(uart_buf, strlen(uart_buf));
 }
 
-void getAverage(void) {
-	uint32_t temp = gRollingSum;
-	temp /= ROLLING_AVG_SIZE;
-	gThrottleOut = temp;
-}
+int main(void) {
+    can_init(BAUD_500KBPS);
+    adc_init();
+    timer_init(&timer0_cfg);
 
+    gpio_set_mode(DEBUG_LED1, OUTPUT);
+    gpio_set_mode(DEBUG_LED2, OUTPUT);
+    gpio_set_mode(DEBUG_LED3, OUTPUT);
+    gpio_set_mode(LED1, OUTPUT);
+    gpio_set_mode(LED2, OUTPUT);
 
-//*************Testing*****************
-void printThrottle1(void) {
-	char uart_buf[64];
-	sprintf(uart_buf, "tout: %d", gThrottleOut);
-	LOG_println(uart_buf, strlen(uart_buf));
+    gpio_set_mode(SS_ESTOP, INPUT);
+    gpio_set_mode(SS_IS, INPUT);
+    gpio_set_mode(SS_BOTS, INPUT);
 
-	sprintf(uart_buf, "v1: %d", gThrottle1Voltage);
-	LOG_println(uart_buf, strlen(uart_buf));
+    gpio_enable_interrupt(SS_ESTOP);
+    gpio_enable_interrupt(SS_IS);
+    gpio_enable_interrupt(SS_BOTS);
 
-	sprintf(uart_buf, "v2: %d", gThrottle2Voltage);
-	LOG_println(uart_buf, strlen(uart_buf));
+    // Initialize interrupts
+    sei();
 
-	sprintf(uart_buf, " ");
-	LOG_println(uart_buf, strlen(uart_buf));
-}
+    // Initialize motor controller
+    can_motor_controller[4] = 1; // anticlockwise
 
-void printThrottle(void) {
-	char uart_buf[32];
-	sprintf(uart_buf, "t10ticks: %d", throttle10Counter);
-	LOG_println(uart_buf, strlen(uart_buf));
-}
+    mc_msg.data = can_motor_controller;
 
-void testStartup(void) {
-	if (bit_is_set(gFlag, FLAG_MOTOR_ON)) {
-		// PLED1_PORT |= _BV(PLED1);
-		// PLED2_PORT |= _BV(PLED2);
-		// PLED3_PORT |= _BV(PLED3);
-	}
-	else {
-		// PLED1_PORT &= ~_BV(PLED1);
-		// PLED2_PORT &= ~_BV(PLED2);
-		// PLED3_PORT &= ~_BV(PLED3);
-	}
-}
+    can_send(&mc_msg)
 
-void throttleLights(void) {
-	// if (gThrottleOut > 64) {
-	//  PLED1_PORT |= _BV(PLED1);
-	// }
-	// else {
-	//  PLED1_PORT &= ~_BV(PLED1);
-	// }
-
-	// if (gThrottleOut > 128) {
-	//  PLED2_PORT |= _BV(PLED2);
-	// }
-	// else {
-	//  PLED2_PORT &= ~_BV(PLED2);
-	// }
-
-	// if (gThrottleOut > 192) {
-	//  PLED3_PORT |= _BV(PLED3);
-	// }
-	// else {
-	//  PLED3_PORT &= ~_BV(PLED3);
-	// }
-}
+    can_receive(&rx_dashboard_msg, rx_dashboard_filter);
 
-void showError(void) {
-	// uint8_t temp = gError;
-
-	// if (temp % 2 == 1) {
-	//  PLED1_PORT |= _BV(PLED1);
-	// }
-	// else {
-	//  PLED1_PORT &= ~_BV(PLED1);
-	// }
-
-	// temp /= 2;
-	//
-	// if (temp % 2 == 1) {
-	//  PLED2_PORT |= _BV(PLED2);
-	// }
-	// else {
-	//  PLED2_PORT &= ~_BV(PLED2);
-	// }
-
-	// temp /= 2;
-	//
-	// if (temp % 2 == 1) {
-	//  PLED3_PORT |= _BV(PLED3);
-	// }
-	// else {
-	//  PLED3_PORT &= ~_BV(PLED3);
-	// }
-}
+    while (1) {
+        if (run_1ms) {
```

Comment:
You don't have a `run_1ms` variable anywhere. You need to define it and hook it up to a timer interrupt. Lmk if you need help with this.

Reply:
Do you know what the loop frequency should be?

Reply:
Switch to `run_10ms`